src/models/dates/DateAsserter.ts

```diff
@@ -0,0 +1,84 @@
+import { DateType } from './DateFormat';
+import { DateComponents } from './DateSplitters';
+
```

Comment:
I see you did quite a lot of work on managing dates, but couldn't we have used a library to do all of this work for us?